Add -Wno-availability to WARNING_FLAGS when building with clang static analyzer

ddkilzer opened the pull request:
#### f1bae761654520d4a00641286ba72e80462bd0e3
<pre>
Add -Wno-availability to WARNING_FLAGS when building with clang static analyzer
<a href="https://bugs.webkit.org/show_bug.cgi?id=267269">https://bugs.webkit.org/show_bug.cgi?id=267269</a>
&lt;<a href="https://rdar.apple.com/120711320">rdar://120711320</a>&gt;

Reviewed by NOBODY (OOPS!).

* Configurations/CommonBase.xcconfig:
(WARNING_CFLAGS):
- Add $(WK_SANITIZER_WARNING_CFLAGS) to the list of values.
* Configurations/Sanitizers.xcconfig:
(WK_SANITIZER_WARNING_CFLAGS): Add.
(WK_STATIC_ANALYZER_WARNING_CFLAGS_YES): Add.
(WK_STATIC_ANALYZER_WARNING_CFLAGS_INTERNAL_SDK_YES): Add.
- Include -Wno-availability during build-and-analyze.
</pre><!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/f1bae761654520d4a00641286ba72e80462bd0e3

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [  ~~🧪 style~~](https://ews-build.webkit.org/#/builders/38/builds/33526 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 ios~~](https://ews-build.webkit.org/#/builders/48/builds/12296 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 mac~~](https://ews-build.webkit.org/#/builders/14/builds/35446 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/5/builds/36147 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/30429 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| | [  ~~🛠 ios-sim~~](https://ews-build.webkit.org/#/builders/49/builds/14652 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 mac-AS-debug~~](https://ews-build.webkit.org/#/builders/51/builds/9456 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/5/builds/36147 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [  ~~🧪 webkitperl~~](https://ews-build.webkit.org/#/builders/11/builds/34001 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/49/builds/14652 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/14/builds/35446 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/5/builds/36147 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/49/builds/14652 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/14/builds/35446 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/2/builds/37471 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/49/builds/14652 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/14/builds/35446 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/2/builds/37471 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/9287 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/51/builds/9456 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/2/builds/37471 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/11051 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/14/builds/35446 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/9871 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/45/builds/10041 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

emw-apple commented on Configurations/Sanitizers.xcconfig:
```diff
@@ -99,3 +101,8 @@ WK_LIBFUZZER_OTHER_LDFLAGS_YES = -fsanitize-coverage=$(WK_LIBFUZZER_COVERAGE);
+// Clang Static Analyzer
+
+// These are to make open source clang work when building WebKit.
+WK_STATIC_ANALYZER_WARNING_CFLAGS_YES = -Wno-availability;
```
From what I can tell (see my comments in bugzilla), we only need to do this in internal builds. If so, let's change the variable lookup to `$(WK_STATIC_ANALYZER_WARNING_CFLAGS_$(RUN_CLANG_STATIC_ANALYZER)_$(USE_INTERNAL_SDK));` 
and rewrite this to:
```suggestion
WK_STATIC_ANALYZER_WARNING_CFLAGS_YES_YES = -Wno-availability;
```

ddkilzer replied:
Having second thoughts about landing this as I don't like the idea of turning off `-Wavailability` for build-and-analyze with internal SDKs.

ddkilzer replied:
I updated this PR to show what I would have landed, but I'm going to close this as I've changed my mind and I don't think this should land since it's basically an unsupported configuration.

(We may want to add static-analyzer-specific warning flags in the future, so this can be used as a template.)

emw-apple reviewed (approved):
r+ with fixup

Commit:
Add -Wno-availability to WARNING_FLAGS when building with clang static analyzer
https://bugs.webkit.org/show_bug.cgi?id=267269
<rdar://120711320>

Reviewed by NOBODY (OOPS!).

* Configurations/CommonBase.xcconfig:
(WARNING_CFLAGS):
- Add $(WK_SANITIZER_WARNING_CFLAGS) to the list of values.
* Configurations/Sanitizers.xcconfig:
(WK_SANITIZER_WARNING_CFLAGS): Add.
(WK_STATIC_ANALYZER_WARNING_CFLAGS_YES): Add.
(WK_STATIC_ANALYZER_WARNING_CFLAGS_INTERNAL_SDK_YES): Add.
- Include -Wno-availability during build-and-analyze.
